pkg/controller/metrics.go

```diff
@@ -0,0 +1,86 @@
+// Copyright 2023 The Kanister Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package controller
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+
+	kanistermetrics "github.com/kanisterio/kanister/pkg/metrics"
+)
+
+// metrics encapsulates all the prometheus metrics that controller
+// needs to own.
+type metrics struct {
+	actionSetResolutionCounterVec prometheus.CounterVec
+}
+
+const (
+	ACTION_SET_COUNTER_VEC_LABEL_RES         = "resolution"
+	ACTION_SET_COUNTER_VEC_LABEL_RES_SUCCESS = "success"
+	ACTION_SET_COUNTER_VEC_LABEL_RES_FAILURE = "failure"
+)
+
+func getActionTypeBucket(actionType string) string {
+	actionTypeBucket := ""
+	switch actionType {
+	case "backup":
+		actionTypeBucket = "backup"
+	case "restore":
+		actionTypeBucket = "restore"
+	case "delete":
+		actionTypeBucket = "delete"
+	case "pre-backup":
```

Comment:
Here are some action names recommended in public documentation
- backup
- restore
- delete
- backupToServer
- restoreFromServer
- before-backup
- on-success
- on-failure
- pre-restore
- post-restore
- post-restore-failed
- backupPrehook
- backupPosthook

Also, I would add string constants instead of using them as raw strings.

Reply:
Thank you!